Drivers/Periphs/Inc/radio.h

```diff
@@ -0,0 +1,67 @@
+#ifndef RADIO_H
+#define RADIO_H
+
+#include "ethernetif.h"
+#include "tcpip.h"
+#include "pbuf.h"
+#include "netif.h"
+#include "dhcp.h"
+#include "stm32f4xx.h"
+#include <string.h>
+//Data types sent across Ethernet will be similar to CAN data structure so it is easier to understand
+//Enum for ID's of all messages that can be sent across CAN bus
+typedef enum {
+    TRIP = 0x02,
+    ALL_CLEAR = 0x101,
+    CONTACTOR_STATE = 0x102,
+    CURRENT_DATA = 0x103,
+    VOLT_DATA = 0x104,
+    TEMP_DATA = 0x105,
+    SOC_DATA = 0x106,
+    WDOG_TRIGGERED = 0x107,
+    CAN_ERROR = 0x108,
+    MOTOR_DISABLE = 0x10A,
+    CHARGE_ENABLE = 0x10C
+} CANId_t;
+
+//Union of data that can be sent across CAN bus. Only one field must be filled out
+typedef union {
+	uint8_t b;
+	uint16_t h;
+	uint32_t w;
+	float f;
+} CANData_t;
+
+/**
+ * @note    idx is only used when an array of data needs to be sent.
+ * @note    data is a union so only one of the fields should be filled out or bad stuff will happen.
+ */
+typedef struct {
+	uint8_t idx : 8;
+	CANData_t data;
+} CANPayload_t;
+
+/*This data type is used to push messages onto the queue*/
+typedef struct {
+    CANId_t id;
+    CANPayload_t payload;
+}CANMSG_t;
+
+//THESE ARE IN HERE FOR NOW BUT SHOULD BE IN CANbus.h !!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
Can you just make CANbus.h? Maybe in another pull request with the rest of the CAN driver, since this probably won't be merged for a while because of issues with the ethernet hardware on the board